spec/views/idv/shared/_warning.html.erb_spec.rb

```diff
@@ -0,0 +1,69 @@
+require 'rails_helper'
+
+describe 'idv/shared/_warning.html.erb' do
+  let(:sp_name) { nil }
+  let(:options) { nil }
+  let(:heading) { 'Warning' }
+  let(:action) { nil }
+
+  before do
+    decorated_session = instance_double(ServiceProviderSessionDecorator)
+    allow(decorated_session).to receive(:sp_name).and_return(sp_name)
```

Comment:
we can combine these two lines:

```suggestion
    decorated_session = instance_double(ServiceProviderSessionDecorator, sp_name: sp_name)
```

(and in the other files, too)

Reply:
Nice 👍  Updated in 78875c7.